services/src/main/java/org/keycloak/services/resources/admin/IdentityProviderResource.java

```diff
@@ -241,6 +243,18 @@ private static void updateUsersAfterProviderAliasChange(Stream<UserModel> users,
         });
     }
 
+    private static void updateUsersAfterProviderDeleted(RealmModel realm, KeycloakSession session, String identityProviderAlias){
+        realm.getClientsStream()
+                .forEach(c->{
+                    session.sessions()
+                            .getUserSessionsStream(realm,c)
+                            .forEach(userSessionModel -> {
+                                FederatedIdentityModel federatedIdentityModel=session.users().getFederatedIdentity(realm,userSessionModel.getUser(),identityProviderAlias);
+                                if(federatedIdentityModel!=null) session.sessions().removeUserSession(realm,userSessionModel);
+                            });
+                });
+    }
```

Comment:
Not sure if this is the best way to get all user sessions associated with a broker - @pedroigor what do you think?

Reply:
Yeah, this won't scale well. Instead, we can introduce additional checks when validating the user/client sessions to check if they originated from authenticating through a broker that no longer exists.

The main point being to avoid iterating over all sessions and eventually invalidating them when the user is interacting with the server.

Reply:
Ok. I'll work on it.